src/OpenTelemetry/Metrics/MetricAggregators/HistogramMetricAggregator.cs

```diff
@@ -22,17 +22,76 @@ namespace OpenTelemetry.Metrics
 {
     internal class HistogramMetricAggregator : IHistogramMetric, IAggregator
     {
+        private static readonly double[] DefaultBoundaries = new double[] { 0, 5, 10, 25, 50, 75, 100, 250, 500, 1000 };
+
         private readonly object lockUpdate = new object();
-        private List<HistogramBucket> buckets = new List<HistogramBucket>();
+        private HistogramBucket[] buckets;
+
+        private double[] boundaries;
 
         internal HistogramMetricAggregator(string name, string description, string unit, Meter meter, DateTimeOffset startTimeExclusive, KeyValuePair<string, object>[] attributes)
+            : this(name, description, unit, meter, startTimeExclusive, attributes, DefaultBoundaries)
+        {
+        }
+
+        internal HistogramMetricAggregator(string name, string description, string unit, Meter meter, DateTimeOffset startTimeExclusive, KeyValuePair<string, object>[] attributes, double[] boundaries)
         {
             this.Name = name;
             this.Description = description;
             this.Unit = unit;
             this.Meter = meter;
             this.StartTimeExclusive = startTimeExclusive;
             this.Attributes = attributes;
+
+            if (boundaries.Length == 0)
+            {
+                boundaries = DefaultBoundaries;
+            }
+            this.boundaries = boundaries;
+            this.buckets = this.InitializeBucket(boundaries);
+        }
+
+        internal HistogramBucket[] InitializeBucket(double[] boundaries)
+        {
+            var buckets = new HistogramBucket[boundaries.Length + 1];
+
+            var lastBoundary = boundaries[0];
+            for (int i = 0; i <= boundaries.Length; i++)
```

Comment:
<= intentional?

Reply:
Yes! We need 1 more than the size of the boundaries.

Reply:
But, I'll refactor to make it clearer!